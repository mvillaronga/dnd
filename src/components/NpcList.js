import React from 'react'
import PropTypes from 'prop-types'
import Npc from './Npc'

const NpcList = ({ npcs, onNpcClick }) => (
  <ul>
    {npcs.map((npc, index) => (
      <Npc key={index} {...npc} onClick={() => onNpcClick(index)} />
    ))}
  </ul>
)

NpcList.propTypes = {
  npcs: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired
    }).isRequired
  ).isRequired,
  onNpcClick: PropTypes.func.isRequired
}

export default NpcList