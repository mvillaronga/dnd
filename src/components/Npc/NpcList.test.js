import React from 'react';
import NpcList from './NpcList';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';

const setup = () => {
  const props = {
    npcs: [
      {name: "bobo", selected: false, id: 1},
      {name: "william", selected: true, id: 2}
    ],

    onNpcClick: jest.fn()
  }

  const render = renderer.create(
    <NpcList {...props} />,
  );

  const full = mount(
    <NpcList {...props} />
  )

  return {props, render, full}
}

configure({adapter: new Adapter()});
describe('NpcList component tests', () => {
  it('Default Spell Card Rendering', () => {
    const {render} = setup()

    expect(render.toJSON()).toMatchSnapshot();
  });

  it('On Click handler', () => {
    const {props, full} = setup()

    full.find("li").first().simulate('click')

    expect(props.onNpcClick).toHaveBeenCalled()
  })
})
