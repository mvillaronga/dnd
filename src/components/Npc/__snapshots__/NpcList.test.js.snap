// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NpcList component tests Default Spell Card Rendering 1`] = `
<ul>
  <li
    onClick={[Function]}
    style={
      Object {
        "color": "blue",
      }
    }
  >
    bobo
  </li>
  <li
    onClick={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
    william
  </li>
</ul>
`;
